### PR TITLE
MM-46398 : Calculate team unread in team sidebar from channels data

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
Previously the unreads for team sidebar was calculated based on teammembership. This pr changed it to calculate based on channel membership instead.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-46398

#### Related Pull Requests
n/a

#### Screenshots
n/a

#### Release Note

```release-note
Teams unread are now calculated based on channels membership and threads only. Team membership is not taken into account.
```

### CODE REVIEW

**Comment by M-ZubairAhmed** [comment]
Pending logic verification. Post that will add unit test for the selector.

**Comment by M-ZubairAhmed** [comment]
🧪 E2E report 
https://automation-dashboard.vercel.app/cycles/11e03f1a-f954-494e-b7a1-2913e9db9eb5

**Review by koox00 — APPROVED** [review]

File: packages/mattermost-redux/src/selectors/entities/channels.ts [hunk 1]
```diff
@@ -666,6 +667,99 @@ export const getUnreadStatus: (state: GlobalState) => BasicUnreadStatus = create
...
+    isCollapsedThreadsEnabled,
+    getThreadCounts,
+    (
+        channels,
+        channelMemberships,
+        channelMessageCounts,
+        users,
+        currentUserId,
+        collapsedThreads,
+        teamthreadCounts,
```
Comment (hmhealey) [inline packages/mattermost-redux/src/selectors/entities/channels.ts:691]: Slight nitpicking, but would you mind changing the names of these two? `collapsedThreads` makes it sound a bit too much like an object, so `collapsedThreadsEnabled` or `crtEnabled` would be a bit clearer, and the capitalization of `teamthreadCounts` is incorrect

**Review by hmhealey — CHANGES_REQUESTED** [review]

File: packages/mattermost-redux/src/selectors/entities/channels.ts (hunk 1 above)
Comment (M-ZubairAhmed) [reply packages/mattermost-redux/src/selectors/entities/channels.ts]: Yes changed

**Comment by jgilliam17** [comment]
Thank you @M-ZubairAhmed 
Tested, looks good to merge.
Verified team unreads - working as before.
[E2E report ](https://mattermost-cypress-report.s3.amazonaws.com/230893-60a6569-onprem-ent-webapp-pr-10946-mm-ee-test:60a6569/mochawesome.html)looks good, no PR related failures.

**Review by jgilliam17 — APPROVED** [review]

**Review by hmhealey — APPROVED** [review]

**Pull request merged**